### PR TITLE
[DX] [Docs] Adding more typing to FieldType and in doc

### PR DESCRIPTION
**Pull request by Roshyo**
I don't think it is a BC break introduction since the DocBlock was already announcing the type. But I'm not sure

### CODE REVIEW

**Review by loic425 — DISMISSED** [review]
I agree with the documentation part, but for the render typehint, I'm pretty sure it is bc-break.

**Comment by Roshyo** [comment]
In the interface, probably indeed :thinking: But we can still add that to the final classes imo.

File: src/Component/FieldTypes/FieldTypeInterface.php [hunk 1]
```diff
@@ -23,10 +23,8 @@ interface FieldTypeInterface
      * $options.
      *
      * @param mixed $data
-     *
-     * @return string
      */
-    public function render(Field $field, $data, array $options);
+    public function render(Field $field, $data, array $options): string;
```
Comment (Zales0123) [inline src/Component/FieldTypes/FieldTypeInterface.php:27]: I believe indeed it is a BC-break :( I would skip that for now, unfortunately. There is a part of me that says "let's just document it and move on with the modern syntax" but I'm not sure... cc @lchrusciel I need a second opinion 🖖

Comment (lchrusciel) [reply src/Component/FieldTypes/FieldTypeInterface.php:27]: I must agree :/ Perhaps we can sooner than later just release GrudBundle 2.0

Comment (Roshyo) [reply src/Component/FieldTypes/FieldTypeInterface.php]: Removed. But I believe we can leave the extending classes with that return type, can we ?

**Review by Zales0123 — APPROVED** [review]

**Comment by Zales0123** [comment]
Thanks, Stephane! :tada:

**Pull request merged**